providers/rfc2136.go

```diff
@@ -0,0 +1,182 @@
+package providers
+
+import (
+	"github.com/Sirupsen/logrus"
+	rdns "github.com/rancher/external-dns/dns"
+	"os"
+	"fmt"
+	"github.com/miekg/dns"
+	"strings"
+	"net"
+	"time"
+)
+
+type RFC2136Handler struct {
+	server     string
+	port       string
+	keyname    string
+	key        string
+	rootdomain string
+}
+
+/* Private functions */
+func needFQDN(domain string) string {
```

Comment:
There is a `dns.Fqdn(domain)` utility function in the `miekg/dns` package doing just that.